### ISSUE TITLE
Event Pattern Matching

### ISSUE THREAD

**Issue by nason** [rubenkaiser/serverless-offline-eventBridge#12]
Hi there, thanks for this great plugin! I'm running into an issue with eventbridge subscriptions that use pattern matching, ie in my `serverless.ts`:

```ts
'created-customer': {
  handler: 'handlers/events/created.ts',
  events: [
    {
      eventBridge: {
        eventBus: '${cf:graph-event-ingest-${self:provider.stage}.EventBusArn}',
        pattern: {
          source: ['graph'],
          'detail-type': ['insert', 'manual'],
          detail: {
            trigger: {
              // @ts-ignore: this is a valid pattern
              name: ['company'],
            },
          },
        },
      },
    },
  ],
},
```

(`@types/serverless` complains about the pattern, but serverless deploys it fine and is a valid [event pattern](https://docs.aws.amazon.com/eventbridge/latest/userguide/filtering-examples-structure.html) -- this part is working)

When I run `sls offline` and make a request, I get the following error though:

```
  TypeError: subscriber.event.pattern.detail[key].includes is not a function
      at /Users/michaelnason/Code/flavortown/node_modules/serverless-offline-aws-eventbridge/src/index.js:199:50
```

Thats brings me to this check: https://github.com/rubenkaiser/serverless-offline-eventBridge/blob/da736dc70dc7e5c9ded49b113bf04822ab4f3c00/src/index.js#L198-L200

Where `key` is `'trigger'` and `subscriber.event.pattern.detail[key]` is `{ name: [ 'company' ] }`

Removing the `detail` part of the pattern gets us around this error, but we use this type of matching pretty heavily in our subscriptions, and would love to be able to test offline more thoroughly. Do you have any suggestions?

**Comment by rubenkaiser** [comment]
Hi, you're welcome. Had a quick look on my mobile and you are right. The AWS docs also state that your config is valid. The check you are referring to (.includes) should be replaced and check the type first (array or object). I will provide a fix in the coming days and keep you posted.

**Comment by rubenkaiser** [comment]
Hi, just published a beta release with a quick fix to see if it will work in your project. In this case the plugin only handles detail patterns one level deeper than previously. If this works I'll update the plugin later with a more robust solution that can handle all kinds of objects and depts.

https://www.npmjs.com/package/serverless-offline-aws-eventbridge/v/1.4.0-beta.0

**Comment by nason** [comment]
Hi @rubenkaiser thanks for the quick response. The beta version does work with the sample I posted!

Some of our other subscriptions are more nested, and those show the same issue. For example:

```ts
{
  source: ['graph'],
  'detail-type': ['update'],
  detail: {
    trigger: {
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore: this is a valid pattern
      name: ['company'],

      event: {
        data: {
          new: {
            status: ['archived'],
          },
        },
      },
    },
  },
},
```

Please let me know if there is any way I can help!

**Comment by rubenkaiser** [comment]
Hi @nason. Good to see that the first test worked for you. I've just release a new beta version in which I had a go to fixing the more nested subscriptions. In this case I decided to flatten the subscription pattern, as well as the event detail. After that I've check if the event has the right properties.

 #13 

I think this is a nice generic solution that works very well. Can you check it in your setup?

https://www.npmjs.com/package/serverless-offline-aws-eventbridge/v/1.4.0-beta.2

**Comment by rubenkaiser** [comment]
Fixed by #13
